app/controllers/users_controller.rb

```diff
@@ -1,12 +1,16 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by(id: params[:id])
-    return if @user
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :load_user, only: [:show, :edit, :update, :destroy,
+                 :correct_user]
+  before_action :admin_user, only: :destroy
 
-    flash[:danger] = t "home.nil_user"
-    redirect_to root_path
+  def index
+    @users = User.all.page(params[:page]).per(Settings.page_num)
```

Comment:
Bản thân cái .page nó là scope rồi. K cần gọi qua .all 
```suggestion
    @users = User.page(params[:page]).per(Settings.page_num)
```